### ISSUE TITLE
How to configure Tracking Protection for suspected fingerprinters?

### ISSUE THREAD

**Issue by htcfreek** [mozilla/policy-templates#1149]
**How can I manage the settings for suspected fingerprinters of the tracking protection?**
**Is there a updated policy? Because setting the preferences is not allowed.**

_Affected preferences:  `privacy.fingerprintingProtection` & `privacy.fingerprintingProtection.pbmode`_

![image](https://github.com/user-attachments/assets/10984e01-9c0e-48df-8bb5-4a67d5c9dd4a)

---

FF version: 128.0.1 ESR

**Comment by mkaply** [comment]
There's a tracking protection policy to configure those

https://mozilla.github.io/policy-templates/#enabletrackingprotection

**Comment by htcfreek** [comment]
Is this a new single policy or do you mean the policies in the "Tracking Protection" folder/category? Because we enabled all policies (except "Exceptions" policy) in the tracking protection folder.

And as you can see all parts except the suspected fingerprinters are working as expected.

(Maybe there is another preference needed or only locking the ui does not work correctly.)

**Comment by htcfreek** [comment]
@mkaply 
Yes this policy exists. But as from the docs it does not manage the required preference `privacy.fingerprintingProtection` & `privacy.fingerprintingProtection.pbmode`.

**Comment by mkaply** [comment]
Good callout. I didn't know the difference.

I'm wondering if I should just add privacy.fingerprintingProtection & privacy.fingerprintingProtection.pbmode to the existing Fingerprinting policy.

**Comment by htcfreek** [comment]
> Good callout. I didn't know the difference.
> 
> I'm wondering if I should just add privacy.fingerprintingProtection & privacy.fingerprintingProtection.pbmode to the existing Fingerprinting policy.

Not sure. But `privacy.fingerprintingProtection` enables it and `privacy.fingerprintingProtection.pbmode` defines the behavior it uses. So maybe it requires a new policy with selection of "off", "on, all tabs" or "on, private browsing".

And don't forget to update the "lock ui" policy as it doesn't work too.

In the first step it would be great if we can set it using the preferences policy.

**Comment by mkaply** [comment]
For the policies, I pretty much ignore pbmode and do all or nothing for all the tracking protection. 

https://searchfox.org/mozilla-central/source/browser/components/enterprisepolicies/Policies.sys.mjs#1092

I'd really rather just add this to the existing policy. I think I'm going to add something called "FingerprintingAttempts" that controls this pref.

**Comment by htcfreek** [comment]
@mkaply 
Can you estimate a time frame for when we can expect this to be solved in an ESR version?

**Comment by nyabinary** [comment]
bump cc @mkaply

**Comment by mkaply** [comment]
I don't know why this took me so dang long.

https://bugzilla.mozilla.org/show_bug.cgi?id=1971182

This should be in the 140 ESR (and I'll look to backport to 128)
